src/cpp/rtps/transport/shared_mem/SharedMemManager.hpp

```diff
@@ -204,7 +376,22 @@ class SharedMemManager
             // Init the segment node
             segment_node_ = segment_->get().construct<SegmentNode>("segment_node")();
             segment_node_->ref_count.exchange(1);
-            segment_node_->free_bytes.exchange(payload_size);
+            segment_node_->pad = 0;
+
+            free_bytes_ = payload_size;
+
+            // Alloc the buffer nodes
+            auto buffers_nodes = segment_->get().construct<BufferNode>
+                        (boost::interprocess::anonymous_instance)[max_allocations]();
+
+            // All buffer nodes are free
+            for (uint32_t i = 0; i<max_allocations; i++)
```

Comment:
```suggestion
            // All buffer nodes are free
            allocated_buffers_.reserve(max_allocations);
            for (uint32_t i = 0; i < max_allocations; i++)
```

Reply:
allocated_buffers_ is an std::list, it doesn't have reserve operator.

Reply:
You're right! Could we convert both lists to vectors and use reserve? 
Or perhaps implement our own list as we did with `semaphores_pool_` on #1147?

Reply:
The problem I see here is that the order of the buffer return is not fifo. So although it is probable that the first free buffers are at the begining (oldest) sometimes you have to delete in the middle, so thats why I use list vs vectors, remove middle elements in a vector is costly.

Reply:
OK. We'll leave it for a separate PR